internal/kustomize/kustomize_test.go

```diff
@@ -0,0 +1,55 @@
+package kustomize
+
+import (
+	"testing"
+
+	argoappv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildKustomizeOptions(t *testing.T) {
+	testCases := []struct {
+		name           string
+		enableHelm     bool
+		loadRestrictor string
+		expected       *argoappv1.KustomizeOptions
+	}{
+		{
+			name:           "Enable helm, no load restrictor",
+			enableHelm:     true,
+			loadRestrictor: "",
+			expected: &argoappv1.KustomizeOptions{
+				BuildOptions: "--enable-helm",
+			},
+		},
+		{
+			name:           "Disable helm, provide load restrictor",
+			enableHelm:     false,
+			loadRestrictor: "LoadRestrictionsNone",
+			expected: &argoappv1.KustomizeOptions{
+				BuildOptions: "--load-restrictor LoadRestrictionsNone",
+			},
+		},
+		{
+			name:           "Disable helm, no load restrictor",
+			enableHelm:     false,
+			loadRestrictor: "",
+			expected: &argoappv1.KustomizeOptions{
+				BuildOptions: "",
+			},
+		},
+		{
+			name:           "Enable helm, provide load restrictor",
```

Comment:
I would add a test with some other `loadRestrictor` value, to demonstrate we pass the value as-is.